templates/testpmd/resource-specs/sriov-vm1-template.yaml

```diff
@@ -0,0 +1,131 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: dpdk-vm-template
+objects:
+- apiVersion: kubevirt.io/v1
+  kind: VirtualMachine
+  metadata:
+    annotations:
+      openshift.io/display-name: "RHEL VM with SR-IOV-backed NIC and DPDK."
+      description: >-
+        Template for RHEL 8.4 VM, with a secondary interface backed by an SR-IOV node NIC.
+        Involves replacing the default NIC driver (iavf) to vfio-pci, to enable running DPDK.
+      openshift.io/documentation-url: "https://github.com/RHsyseng/cnv-supplemental-templates/tree/main/templates/testpmd"
+    name: ${NAME}
+    namespace: ${NAMESPACE}
+    labels:
+      app: ${NAME}
+      vm.kubevirt.io/template: dpdk-vm-template
+  spec:
+    dataVolumeTemplates:
+    - apiVersion: cdi.kubevirt.io/v1beta1
+      kind: DataVolume
+      metadata:
+        name: ${NAME}
+      spec:
+        source:
+          registry:
+            pullMethod: node
+            url: ${IMAGE_URL}
+        storage:
+          resources:
+            requests:
+              storage: ${DV_SIZE}
+    running: false
+    template:
+      spec:
+        domain:
+          cpu:
+            cores: ${{NUM_OF_CORES}}
+          devices:
+            disks:
+            - disk:
+                bus: virtio
+              name: ${NAME}
+            - disk:
+                bus: virtio
+              name: cloudinitdisk
+            interfaces:
+            - masquerade: {}
+              name: default
+            - macAddress: ${SECONDARY_MAC}
+              name: sriov-test-network
+              sriov: {}
+            rng: {}
+          memory:
+            guest: ${MEMORY}
```

Comment:
Is it a must? inst it enough to specify resources.requests.memory?

Reply:
It is a must, we tried it without it and the VM failed to start (I actually don't remember the exact error).

Reply:
As far as I understand memory.guest is used for over committing memory (let the VM know it has more memory then the node can actually provide) check this out https://kubevirt.io/user-guide/operations/node_overcommit/#node-overcommit
Did I got it wrong?
Is there a chance it failed on your setup due to lake of resources?

Reply:
From the experience of other resembling templates (in terms of memory requirements), we learned that we should specify both `memory.guest` and `resources.requests.memory`:
`memory.guest` is for the guest VM - that's what it "knows" it has for use, while `resources.requests.memory` is an extra memory, which is to be used by the pod.
Hope that clarifies.